### PR TITLE
Remove legacy time spine warning message

### PR DESCRIPTION
**Pull request by DevonFulcher**
Resolves # SL-2934

### Description

This PR removes the legacy time spine configuration warning in order to make the warning appear in dbt-core as a [behavior change](https://docs.getdbt.com/reference/global-configs/behavior-changes) instead.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: tests/validations/test_time_spines.py [hunk 1]
```diff
@@ -75,38 +74,6 @@ def test_valid_time_spines() -> None:  # noqa: D
...
-        metrics=[],
-        project_configuration=PydanticProjectConfiguration(
-            time_spine_table_configurations=[
-                PydanticTimeSpineTableConfiguration(location="hurrr", column_name="fun_col", grain=TimeGranularity.DAY)
-            ]
-        ),
-    )
-    issues = validator.validate_semantic_manifest(semantic_manifest)
-    assert not issues.has_blocking_issues
-    assert len(issues.warnings) == 1
```
Comment (courtneyholcomb) [inline tests/validations/test_time_spines.py:106]: Could keep this test but update this to say `assert len(issues.warnings) == 1` and remove the line below!

**Review by courtneyholcomb — APPROVED** [review]
Looks good - note that we'll need to release a new production version in order to ship this to core. That process is documented here https://www.notion.so/dbtlabs/DSI-Build-Release-Process-48d94828e8b840abb8529f7501fb2c5b?pvs=4

**Comment by courtneyholcomb** [comment]
Actually, we shouldn't do the production release until after code freeze, since mantle could pick up the new version.

File: tests/validations/test_time_spines.py (hunk 1 above)
Comment (DevonFulcher) [reply tests/validations/test_time_spines.py:106]: I assume you mean `assert len(issues.warnings) == 0`, right?

Comment (courtneyholcomb) [reply tests/validations/test_time_spines.py:106]: Yes 😆

**Pull request merged**